src/main/java/com/glencoesoftware/omero/ms/image/region/ImageRegionCtx.java

```diff
@@ -137,11 +143,43 @@ public void assignParams(MultiMap params) throws IllegalArgumentException {
             this.maps = Json.decodeValue(maps, List.class);
         }
         format = Optional.ofNullable(params.get("format")).orElse("jpeg");
+        cacheKey = createCacheKey(params);
 
         log.debug(
                 "{}, z: {}, t: {}, tile: {}, c: [{}, {}, {}], m: {}, " +
-                "format: {}", imageId, z, t, tile, channels, windows, colors,
-                m, format);
+                "format: {}, flip: {}", imageId, z, t, tile, channels, windows,
+                colors, m, format, flip, cacheKey);
+    }
+
+    /**
+     * Creates a cache key for the context using the class name and originally
+     * provided parameter map.
+     * @param params {@link io.vertx.core.http.HttpServerRequest} parameters
+     * required for rendering an image region.
+     * @return A cache key as outlined above using the
+     * <a href="https://131002.net/siphash/">64-bit SipHash-2-4 algorithm</a>
+     * and default Guava seed value.
+     * @see {@link com.google.common.hash.Hashing#sipHash24()}
+     */
+    private String createCacheKey(MultiMap params) {
```

Comment:
@kevin: This loops through entries in the `MultiMap` of URL parameters to build the cache key. A render image region request where `...?a=1&b=2` should have the same cache key as `...?b=2&a=1`. Does it? In looking at the `MultiMap` documentation it seemed undefined to me what the key/entry order is.